Software/app/src/main/java/org/foi/emp/collegecoursemanager/DodavanjeBodovaKolegiju.java

```diff
@@ -1,20 +1,102 @@
+
+    @RequiresApi(api = Build.VERSION_CODES.N)
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_dodavanje_bodova_kolegiju);
         Intent i = getIntent();
-        if (i != null && i.getIntExtra("ID_kolegija", 0) != 0) {
-            Toast.makeText(getApplicationContext(), "Ko kumi", Toast.LENGTH_SHORT).show();
+        this.ostvareniBodovi = findViewById(R.id.txtOstvareniBodovi);
+        this.ostvareniBodovi.setInputType(InputType.TYPE_CLASS_NUMBER);
+        this.spListaElemenataKolegija = findViewById(R.id.spElementiModelaPracenja);
+        this.spremiBodove = findViewById(R.id.btnSpremiBodove);
+        this.nazivKolegija = findViewById(R.id.tvNazivKolegijaZaBodove);
+        if (i != null && i.getIntExtra(INTENT_KOLEGIJ_ID, 0) != 0) {
+            Kolegij kolegij = Database.getInstance(getApplicationContext()).getKolegijDAO().dohvatiKolegij(i.getIntExtra(INTENT_KOLEGIJ_ID, 0));
+            elementiModelaPracenja = DohvatiListuElemenataModelaPracenja(kolegij);
+            //Database.getInstance(getApplicationContext()).getModelPracenjaDAO().dohvatiListuElemenataModelaPracenjaPoModeluPracenjaLIVE(kolegij.getModelPracenja());
+
+            this.nazivKolegija.setText(kolegij.getNazivKolegija());
+            PostaviSpinner();
+            spListaElemenataKolegija.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+                @Override
+                public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                    ElementModelaPracenja emp = elementiModelaPracenja.get(position);
+                    ostvareniBodovi.setText(String.valueOf(emp.getOstvareniBodovi()));
+                    spremiBodove.setOnClickListener(new View.OnClickListener() {
+                        @Override
+                        public void onClick(View v) {
+                            if(Integer.parseInt(ostvareniBodovi.getText().toString())<=emp.getMaksimalniBrojBodova()){
+                                emp.setOstvareniBodovi(Integer.parseInt(ostvareniBodovi.getText().toString()));
+                                Database.getInstance(getApplicationContext()).getModelPracenjaDAO().ažuriranjeElementaModelaPracenja(emp);
+                                Toast.makeText(getApplicationContext(),"Uspjesno unesni bodovi za "+emp.getNaziv(),Toast.LENGTH_SHORT).show();
+                            }
+                            else{
+                                Toast.makeText(getApplicationContext(),"Ostvareni bodovi ne mogu biti veci od "+String.valueOf(emp.getMaksimalniBrojBodova()),Toast.LENGTH_SHORT).show();
+                            }
+                        }
+                    });
+                }
+                @Override
+                public void onNothingSelected(AdapterView<?> parent) {
+                }
+            });
```

Comment:
This looks a little bit messy and unreadable. Could you put this too an function?
Idea is to make code readable, modular and not complex. You're having 3-4 functions inside function which is hard to follow and now the idea behind. Try not to do it in following. 
```
public void setListenerOnSpinner(fuction formal parameters){
            spListaElemenataKolegija.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {                
            @Override                
            public void onItemSelected(AdapterView&lt;?&gt; parent, View view, int position, long id) {                                                
                ElementModelaPracenja emp = elementiModelaPracenja.get(position);                        
                ostvareniBodovi.setText(String.valueOf(emp.getOstvareniBodovi()));                        
                spremiBodove.setOnClickListener(new View.OnClickListener() {                        
                @Override                        
                public void onClick(View v) {                                    
                    if(Integer.parseInt(ostvareniBodovi.getText().toString())&lt;=emp.getMaksimalniBrojBodova()){                                
                        emp.setOstvareniBodovi(Integer.parseInt(ostvareniBodovi.getText().toString()));                                          
                        Database.getInstance(getApplicationContext())
                                        .getModelPracenjaDAO().ažuriranjeElementaModelaPracenja(emp);                                        
                        Toast.makeText(getApplicationContext(),
                        "Uspjesno unesni bodovi za "+emp.getNaziv(),Toast.LENGTH_SHORT).show();                            
                    }
                    else{
                        Toast.makeText(getApplicationContext(),"Ostvareni bodovi ne mogu biti veci od "+String.valueOf(emp.getMaksimalniBrojBodova()),Toast.LENGTH_SHORT).show();                            
                    }                        
                }                    
            });                
        }                
        @Override                
        public void onNothingSelected(AdapterView&lt;?&gt; parent) {}            
    });
}
```